src/selectors/dataset.js

```diff
@@ -14,16 +14,91 @@ export const datasetSelector = (config, state, instance) => {
   const baseState = config.baseSelector(state)
 
   // Get instance array
-  const instanceData = baseState.getIn(['instances', instance, 'data'])
+  let instanceData = baseState.getIn(['instances', instance, 'data'])
   const rawData = baseState.get('raw')
 
   // If instanceData is not created yet, return empty Map
   if(!instanceData) {
     return Map({})
   }
 
+  // Remove the pending.create items from the instance
+  instanceData = instanceData.filter(uid => baseState.getIn(['pending', 'create'], List()).indexOf(uid) === -1)
```

Comment:
We should only get `baseState.getIn(['pending', 'create'], List()` once above set in a const. That way we aren't getting the pending create items inside a loop.

Something like:
```
const pendingCreate = baseState.getIn(['pending', 'create'], List()
instanceData = instanceData.filter(uid => pendingCreate.indexOf(uid) === -1)
```